### PR TITLE
[ADMNAPI-989] Add read me file and fix docker file warnings

### PR DESCRIPTION
**Pull request by CSR2017**

### CODE REVIEW

File: docs/docker.md [hunk 1]
```diff
@@ -57,12 +57,13 @@ style I fill:#fff
...
    bash ./generate-certificate.sh
    ```
 
 2. Copy and customize the `.env.example` file. Importantly, be sure to change
-   the encryption key.
+   the encryption key. In a Bash prompt, generate a random key thusly: `openssl
+   rand -base64 32`.
 
    ```shell
    cd ../../Compose/pgsql
```
Comment (stephenfuqua) [inline docs/docker.md:69]: In Admin API 2, there are two sub-directories here. Probably need to mention that in the documentation.

**Review by stephenfuqua — COMMENTED** [review]
I was unable to test locally, even running on my home network which usually works for Docker containers. I get an error when the container reaches the `dotnet restore` command. This is probably just something weird happening on my computer. Would like to see someone else test this.

```
 => ERROR [adminapi build 6/6] RUN dotnet restore && dotnet build -c Release                                          2.0s
------
 > [adminapi build 6/6] RUN dotnet restore && dotnet build -c Release:
0.888   Determining projects to restore...
1.967 /usr/share/dotnet/sdk/6.0.201/NuGet.targets(130,5): error : Unable to load the service index for source https://api.nuget.org/v3/index.json. [/source/EdFi.Ods.AdminApi/EdFi.Ods.AdminApi.csproj]
1.967 /usr/share/dotnet/sdk/6.0.201/NuGet.targets(130,5): error :   Name does not resolve (api.nuget.org:443) [/source/EdFi.Ods.AdminApi/EdFi.Ods.AdminApi.csproj]
1.967 /usr/share/dotnet/sdk/6.0.201/NuGet.targets(130,5): error :   Name does not resolve [/source/EdFi.Ods.AdminApi/EdFi.Ods.AdminApi.csproj]
```

File: Docker/api.Dockerfile
Comment (stephenfuqua) [inline Docker/api.Dockerfile:1]: This file is still using an old base images, both the dotnet/sdk and dotnet/aspnet images need to be updated.

**Review by stephenfuqua — COMMENTED** [review]
Please also update `dev.Dockerfile` with the same changes that we have in https://github.com/Ed-Fi-Alliance-OSS/AdminAPI-1.x/blob/main/Docker/dev.Dockerfile

**Comment by CSR2017** [comment]
> I was unable to test locally, even running on my home network which usually works for Docker containers. I get an error when the container reaches the `dotnet restore` command. This is probably just something weird happening on my computer. Would like to see someone else test this.
> 
> ```
>  => ERROR [adminapi build 6/6] RUN dotnet restore && dotnet build -c Release                                          2.0s
> ------
>  > [adminapi build 6/6] RUN dotnet restore && dotnet build -c Release:
> 0.888   Determining projects to restore...
> 1.967 /usr/share/dotnet/sdk/6.0.201/NuGet.targets(130,5): error : Unable to load the service index for source https://api.nuget.org/v3/index.json. [/source/EdFi.Ods.AdminApi/EdFi.Ods.AdminApi.csproj]
> 1.967 /usr/share/dotnet/sdk/6.0.201/NuGet.targets(130,5): error :   Name does not resolve (api.nuget.org:443) [/source/EdFi.Ods.AdminApi/EdFi.Ods.AdminApi.csproj]
> 1.967 /usr/share/dotnet/sdk/6.0.201/NuGet.targets(130,5): error :   Name does not resolve [/source/EdFi.Ods.AdminApi/EdFi.Ods.AdminApi.csproj]
> ```

@stephenfuqua I had similar issue after updating my docker desktop version. I followed this to adding the dns value temporarily https://github.com/dotnet/core/issues/8048

File: Docker/api.Dockerfile
Comment (CSR2017) [reply Docker/api.Dockerfile:1]: .Net 8 migration changes will be addressed as part of https://github.com/Ed-Fi-Alliance-OSS/AdminAPI-2.x/pull/95

**Comment by CSR2017** [comment]
> Please also update `dev.Dockerfile` with the same changes that we have in https://github.com/Ed-Fi-Alliance-OSS/AdminAPI-1.x/blob/main/Docker/dev.Dockerfile

.Net 8 migration changes will be addressed as part of https://github.com/Ed-Fi-Alliance-OSS/AdminAPI-2.x/pull/95

**Comment by stephenfuqua** [comment]
> @stephenfuqua I had similar issue after updating my docker desktop version. I followed this to adding the dns value temporarily [dotnet/core#8048](https://github.com/dotnet/core/issues/8048)

Thank you for this link. Another solution mentioned there is to install `curl`, which will update the DNS server. I tried that and it worked.

We were already installing `curl` on line 38. We can move that earlier in the code to occur before `dotnet restore`. Move all of the `apk add` commands earlier, so that we can cache as a base layer. End up with these layers:

1. Base layer from Microsoft `AS buildBase`
2. Install curl - new base layer for caching `AS build`
3. Copy in source code - new layer `AS publish`
4. Start from new Microsoft base layer `AS runtimeBase`
5. Install curl and other required packages `AS runtime`
6. Finally, copy the DLL's from layer 3 into layer 4.

**Comment by CSR2017** [comment]
> > @stephenfuqua I had similar issue after updating my docker desktop version. I followed this to adding the dns value temporarily [dotnet/core#8048](https://github.com/dotnet/core/issues/8048)
> 
> Thank you for this link. Another solution mentioned there is to install `curl`, which will update the DNS server. I tried that and it worked.
> 
> We were already installing `curl` on line 38. We can move that earlier in the code to occur before `dotnet restore`. Move all of the `apk add` commands earlier, so that we can cache as a base layer. End up with these layers:
> 
> 1. Base layer from Microsoft `AS buildBase`
> 2. Install curl - new base layer for caching `AS build`
> 3. Copy in source code - new layer `AS publish`
> 4. Start from new Microsoft base layer `AS runtimeBase`
> 5. Install curl and other required packages `AS runtime`
> 6. Finally, copy the DLL's from layer 3 into layer 4.

I tried these steps for rearranging the commands to cache the curl. But still getting the "Unable to load the service index for source" error. I noticed that, we dont have any issue running the Admin API 1.x dev.dockerfile, the difference is - upgraded to .net 8. So I tried running the dev.dockerfile from Johnny's PR. It is working fine. Should I still try accomplish the suggested changes ?

**Comment by stephenfuqua** [comment]
I recommend re-arranging the layers anyway, as this is a better structure. Apparently the error message we are both seeing has different solutions. Moving the curl command earlier fixed it for me, without having to modify the Docker engine configuration.

File: Docker/api.Dockerfile [hunk 2]
```diff
@@ -4,24 +4,29 @@
 # See the LICENSE and NOTICES files in the project root for more information.
 
 #tag 6.0-alpine
-FROM mcr.microsoft.com/dotnet/aspnet@sha256:201cedd60cb295b2ebea7184561a45c5c0ee337e37300ea0f25cff5a2c762538
-LABEL maintainer="Ed-Fi Alliance, LLC and Contributors <techsupport@ed-fi.org>"
-ARG VERSION=latest
+FROM mcr.microsoft.com/dotnet/aspnet@sha256:201cedd60cb295b2ebea7184561a45c5c0ee337e37300ea0f25cff5a2c762538 AS base
```
Comment (stephenfuqua) [inline Docker/api.Dockerfile:7]: I really thought that I saw the api.nuget.org connectivity problem being resolved a few days ago when I just moved the `curl` command. Like you, however, this didn't work for me 😞 .

But... I did find an easy solution: just update the base image to the .NET 8 version! Even though .NET 8 upgrade isn't finished, the SDK for .NET 8 will build a project that targets .NET 6.

Changing to the following solved the problem for me. This is slightly newer than the version we have in Admin API 1 - this version must have been released just a few days ago.

```
FROM mcr.microsoft.com/dotnet/sdk:8.0.203-alpine3.18@sha256:2a8dca3af111071172b1629c12eefaeca0d6c2954887c4489195771c9e90833c as buildBase
```

Comment (stephenfuqua) [reply Docker/api.Dockerfile:7]: A PR into your branch: https://github.com/Ed-Fi-Alliance-OSS/AdminAPI-2.x/pull/97

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**